tools/swarm/src/compose.rs

```diff
@@ -512,34 +533,12 @@ mod tests {
         }
     }
 
-    #[derive(Debug)]
-    struct TestEnv {
-        env: HashMap<String, String>,
-        /// Set of env variables that weren't fetched yet
-        untouched: RefCell<HashSet<String>>,
-    }
-
     impl From<FullPeerEnv> for TestEnv {
         fn from(peer_env: FullPeerEnv) -> Self {
             let json = serde_json::to_string(&peer_env).expect("Must be serializable");
-            let env: HashMap<_, serde_json::Value> =
+            let env: HashMap<_, String> =
```

Comment:
this will start failing if you put any integers in a `.yml` file. It happened for me because at first I used `CHAIN_ID: 0`. That's why I changed it to `serde_json::Value`. I think you should return this

Reply:
First of all, this is a test environment which mocks actual ENV access. It doesn't have anything to do with docker compose yaml.
Secondly, even if docker compose yaml contains some numbers, they will be translated to the system env and read as strings anyway.

Reply:
ok, then it was because I introduced CHAIN_ID=0 env variable which was not a string and this was breaking. You can try yourself, it will break

Reply:
I think it was failing because ENVs were parsed as JSON values. Now they are parsed as they are  - using `FromStr`. Will add a test for it anyway.